src/browser/BrowserAction.cpp

```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright (C) 2023 KeePassXC Team <team@keepassxc.org>
+ *  Copyright (C) 2024 KeePassXC Team <team@keepassxc.org>
```

Comment:
I'll add an option for this that allows localhost for testing purposes.

Reply:
The logic for deciding if something is a secure context is described [here](https://html.spec.whatwg.org/multipage/webappapis.html#secure-context). Ignoring the parts that aren't relevant because they're not allowed origins for WebAuthn, the origins that should be allowed are:

- `https://` followed by anything
- Any origin with a host that is `localhost` or `localhost.`
- Any origin with a host that ends with `.localhost` or `.localhost.`
- `file://` followed by anything

So it would be good if the logic allowed all of them.

This disregards origins that the browser might consider to be authenticated or trustworthy (the spec gives the example of browser extension pages' origins), but I don't think KeePassXC has any way of knowing what those would be and it seems like a niche case anyway.